sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/event_data_batch.hpp

```diff
@@ -0,0 +1,274 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include "eventhub_constants.hpp"
+#include "models/event_data.hpp"
+
+#include <azure/core/amqp/models/amqp_message.hpp>
+#include <azure/core/diagnostics/logger.hpp>
+#include <azure/core/internal/diagnostics/log.hpp>
+
+#include <mutex>
+#include <stdexcept>
+
+// cspell: words vbin
+
+namespace Azure { namespace Messaging { namespace EventHubs {
+
+  /** @brief EventDataBatchOptions contains optional parameters for the
+   * [ProducerClient.CreateEventDataBatch] function.
+   *
+   * @remark If both PartitionKey and PartitionID are nil, Event Hubs will choose an arbitrary
+   * partition for any events in this [EventDataBatch].
+   */
+  struct EventDataBatchOptions
+  {
+
+    /** @brief MaxBytes overrides the max size (in bytes) for a batch.
+     * By default CreateEventDataBatch will use the max message size provided by the service.
+     */
+    uint32_t MaxBytes = std::numeric_limits<int32_t>::max();
+
+    /** @brief PartitionKey is hashed to calculate the partition assignment.Messages and message
+     * batches with the same PartitionKey are guaranteed to end up in the same partition.
+     * Note that if you use this option then PartitionID cannot be set.
+     */
+    std::string PartitionKey;
+
+    /** @brief PartitionID is the ID of the partition to send these messages to.
+     * Note that if you use this option then PartitionKey cannot be set.
+     */
+    std::string PartitionID;
+  };
+
+  /**@brief EventDataBatch is used to efficiently pack up EventData before sending it to Event Hubs.
+   *
+   * @remark EventDataBatch's are not meant to be created directly. Use
+   * [ProducerClient.CreateEventDataBatch], which will create them with the proper size limit for
+   * your Event Hub.
+   */
+  class EventDataBatch {
```

Comment:
```suggestion
  class EventDataBatch final
  {
```

Alternatively - add public virtual or protected non-virtual destructor.